core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3804,6 +3805,38 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleShareGroupHeartbeat(request: RequestChannel.Request): CompletableFuture[Unit] = {
+    val shareGroupHeartbeatRequest = request.body[ShareGroupHeartbeatRequest]
+
+    if (!config.isShareGroupEnabled) {
+      // The API is not supported when the configuration has not been set explicitly
+      requestHelper.sendMaybeThrottle(request, shareGroupHeartbeatRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
+      CompletableFuture.completedFuture[Unit](())
+    } else if (!authHelper.authorize(request.context, READ, GROUP, shareGroupHeartbeatRequest.data.groupId)) {
+      requestHelper.sendMaybeThrottle(request, shareGroupHeartbeatRequest.getErrorResponse(Errors.GROUP_AUTHORIZATION_FAILED.exception))
+      CompletableFuture.completedFuture[Unit](())
+    } else {
+      /* TODO: Once ShareGroupHeartBeatAPI is supported in group coordinator, uncomment the code below
+          (lines 3821 - 3833) and remove the part sending INVALID_CONFIG exception (lines 3835, 3836)
+      groupCoordinator.shareGroupHeartbeat(
+        request.context,
+        shareGroupHeartbeatRequest.data,
+      ).handle[Unit] { (response, exception) =>
+
+        info(s"*** SHARE GROUP HEARTBEAT RESPONSE *** + ${response}")
+
+        if (exception != null) {
+          requestHelper.sendMaybeThrottle(request, shareGroupHeartbeatRequest.getErrorResponse(exception))
+        } else {
+          requestHelper.sendMaybeThrottle(request, new ShareGroupHeartbeatResponse(response))
+        }
+      }
+       */
+      requestHelper.sendMaybeThrottle(request, shareGroupHeartbeatRequest.getErrorResponse(Errors.INVALID_CONFIG.exception))
```

Comment:
since we are already sending Errors.UNSUPPORTED_VERSION in the case when config.isShareGroupEnabled is false ([reference](https://github.com/confluentinc/kafka/pull/1003/files#diff-cc056b4960ededba37a438b1454f0f3c5ff5e8ad5e6d2ec9a08e813ca056ffebR3813)), I used a different error here and from the list of errors, this one seemed more suitable than others. Happy to change if you disagree

Reply:
Doesn't matter much though as the code will be changed shortly, but again it's better to use UNSUPPORTED_VERSION.